test/test_operations.py

```diff
@@ -749,6 +749,12 @@ def test_masked_select_shape(self):
         torch.masked_select(x, mask), 0)
     self.assertEqual(x_dim0_shape.item(), 3)
 
+  def test_nonzero_cast(self):
+    t1 = torch.ones(5, 2, device=xm.xla_device())
+    # this cast should not fail
```

Comment:
mind adding a comment about where the cast happens, such as https://github.com/pytorch/xla/issues/4155#issuecomment-1325562170.?